File: WEB_APPLICATION/BookBox_WebApp/backend/models/Book.js
const mongoose = require('mongoose');

const bookSchema = new mongoose.Schema({
    title: {
        type: String,
        required: true,
        trim: true,
    },
    author: {
        type: String,
        required: true,
    },
    description: {
        type: String,
        trim: true,
    },
    genre: {
        type: String,
        trim: true,
    },
    status: {
        type: String,
        enum: ['available', 'reserved' ,'borrowed'],
        default: 'available',
    },
    owner: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
    },
    parcelLocker: {
        type: mongoose.Schema.Typer.ObjectId,
        ref: 'ParcelLocker',
    },
},
{ timestamps: true }
);

module.exports = mongoose.model('Book', bookSchema);

File: PHONE_APP/app/src/main/java/com/lanteam/bookbox/model/Book.kt
package com.lanteam.bookbox.model

data class Book (
    val id: String ="",
    val title: String,
    val author: String,
    val summary: String,
    val imageUrl: String ="",
    val genre: String ="",
    val status: String ="available", //maybe bool?
    val weight: Int =5,
    val packetBoxId: String? =null,
    var distance: Float?=null,
    val owner:String="",
    val currentBorrower:String?=null
)